### ISSUE TITLE
lint: docStub false positives

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#261]
Reproducer:
```
gocritic check-project -enable=docStub $GOROOT/src
```
Output:
```
cmd/compile/internal/syntax/parser.go:2026:1: docStub: silencing go lint doc-comment warnings is unadvised
```
Code:
```go
// Statement =
// 	Declaration | LabeledStmt | SimpleStmt |
// 	GoStmt | ReturnStmt | BreakStmt | ContinueStmt | GotoStmt |
// 	FallthroughStmt | Block | IfStmt | SwitchStmt | SelectStmt | ForStmt |
// 	DeferStmt .
func (p *parser) stmtOrNil() Stmt {
```
This is not doc comment stub, and it's not exported function at all.
To avoid such false positives, `docStub` should check that function being checked is actually exported (see `ast.IsExported`) and check that there are no more lines inside godoc comment.
Comment above has 5 elements in it's comment group.

Negative tests must be expanded.

**Comment by Quasilyte** [comment]
CC @weeellz, are you interested in this?

**Comment by cristaloleg** [comment]
I can pick up this one :)
